### PR TITLE
Updated edm documentation

### PR DESCRIPTION
**Pull request by krealyt**
Updated the edm documentation page. Harmonized detailed edm docu with Quick Start Guide. Removed deprecated parameters. Open questions have been clarified while updating the documentation in repo everest-dev-environment.

### CODE REVIEW

**Comment by krealyt** [comment]
Is the "--git-pull" parameter a 1:1 replacement for the former "--update" parameter?

Means: Does the following warning also apply to the --git-pull parameter?

> Be advised that even if you remove a repository from the config file it WILL NOT be deleted from the workspace.

**Comment by hikinggrass** [comment]
> Is the "--git-pull" parameter a 1:1 replacement for the former "--update" parameter?
> 
> Means: Does the following warning also apply to the --git-pull parameter?
> 
> > Be advised that even if you remove a repository from the config file it WILL NOT be deleted from the workspace.

That warning does not apply

File: docs/dev_tools/edm.rst [hunk 1]
```diff
@@ -183,8 +171,9 @@ repository. It should look like this:
 	  git_tag: main
 	  options: ["BUILD_EXAMPLES OFF"]
 
-If you want to automatically check out certain dependencies into a workspace,
-you can add a **workspace.yaml** file to the root of your source repository. It should look like this:
+If you want to automatically check out certain dependencies into a workspace
```
Comment (hikinggrass) [inline docs/dev_tools/edm.rst]: I think the section mentioning workspace.yaml files does not apply anymore

**Comment by krealyt** [comment]
> > > Be advised that even if you remove a repository from the config file it WILL NOT be deleted from the workspace.
> 
> That warning does not apply

Deleted the warning.

File: docs/dev_tools/edm.rst (hunk 1 above)
Comment (krealyt) [reply docs/dev_tools/edm.rst]: @hikinggrass Is the whole section then obsolete?
See https://everest.github.io/nightly/dev_tools/edm.html#using-the-edm-cmake-module-and-dependencies-yaml

I assume, only the section with the two code snippets about workspace.yaml is not needed anymore, correct?

Comment (krealyt) [reply docs/dev_tools/edm.rst]: Deleted the reference to workspace.yaml for now. Whole edm docu chapter will be tested and updated with additional changes in another PR in future.

**Pull request merged**